### PR TITLE
[luajit] Use common build functions

### PR DESCRIPTION
**Pull request by dg0yt**
Use existing maintainer functions for config and make.
Build host tools for cross builds as possible.
- The port needs a host tool with the same bitness (32/64) as the target.
- For x86 MSVC, it simply builds the x86 tool with the target toolchain. (ARM64 and UWP are unsupported, so it all stays within a regular Windows system.) Targeting x64 from x86 is unsupported.
- For other platforms, host tools for different platforms are bundled by bitness.
  Originally, building the host tool would require preprocessor output from the target toolchain. This is not possible with vcpkg triplets. This is resolved by hardcoding relevant settings in the generated Makefile, generalized from manual introspection for android, linux, mingw.

...

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
Cross-CPU-builds are hard for this port:
They need a host (triplet) tool `buildvm`, but this tool can be built only for a single target CPU.

**Comment by dg0yt** [comment]
Reference for luajit installation: https://luajit.org/install.html

- Windows MSVC: only x86/x64 at this point
- Non-Windows: x86, x64, arm, arm64
  but ...
  "The GNU Makefile-based [luajit] build system allows cross-compiling on any host for any supported target, as long as both architectures have the same pointer size."
  Given vcpkg approach to require host tools to built for and reused from the host triplet, cross builds in vcpkg can only reach a limited set of target triplets:
  - Reliably those of the same pointer size (x64 to arm64).
  - Maybe unreliably some 64 to 32 builds when the host toolchain can build and run executables for 32 bit (`-m32`).

**Comment by dg0yt** [comment]
The cross-build problem isn't "pointer size" but bootstrapping.
1. host triplet: Get or build some host lua (default: included minilua).
2. target triplet: Generate `host/buildvm_arch.h` based detected target properties.
3. host triplet: Compile `buildvm` host tool.
4. target triplet: Run `buildvm` to generate target asm.

The best thing I can think of is capturing the target properties from step 2 so that `host/buildvm_arch.h` can be generated for any supported cross target.

**Comment by dg0yt** [comment]
Hm, I wonder if the buildvm binaries are really able to work across platforms, i.e. wheter the x64 buildvm built on linux can be used target x64 android.

**Comment by dg0yt** [comment]
> Hm, I wonder if the buildvm binaries are really able to work across platforms, i.e. wheter the x64 buildvm built on linux can be used target x64 android.

No, vcpkg is wrong: https://github.com/microsoft/vcpkg/pull/26617#discussion_r1183270859

File: ports/luajit/configure [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+BUILD_OPTIONS += 'HOST_CC=:'
+BUILD_OPTIONS += 'BUILDVM_T='
+BUILD_OPTIONS += 'BUILDVM_X=${LUAJIT_BUILDVM_X}'
+endif
+
+# used by src/Makefile, best effort from manual '<CC> -E src/lj_arch.h -dM'
+TARGET_TESTARCH_COMMON_64 += LJ_ARCH_BITS 64  LJ_ARCH_HASFPU 1  LJ_LE 1  LJ_HASJIT 1
+TARGET_TESTARCH_COMMON_64 += LJ_HASFFI 1  LJ_ARCH_HASFPU 1  LJ_ABI_SOFTFP 0  LJ_TARGET_GC64 1
+TARGET_TESTARCH_arm64 = \$(TARGET_TESTARCH_COMMON_64)  LJ_TARGET_ARM64  LJ_ARCH_VERSION 80
+TARGET_TESTARCH_x64 =   \$(TARGET_TESTARCH_COMMON_64)  LJ_TARGET_X64  LJ_DUALNUM 1  LJ_FR2 1
```
Comment (dg0yt) [inline ports/luajit/configure:58]: TODO: Provide definitions for x86 and arm

**Review by FrankXie05 — APPROVED** [review]

**Pull request merged**